cl/opinion_page/views.py

```diff
@@ -948,6 +1098,159 @@ async def view_authorities(
     )
 
 
+async def check_flag_exists(flag_name: str) -> bool:
+    return await sync_to_async(
+        waffle.get_waffle_flag_model().objects.filter(name=flag_name).exists
+    )()
+
+
+@never_cache
+async def view_opinion(request: HttpRequest, pk: int, _: str) -> HttpResponse:
+    """View for displaying opinions."""
+
+    flag_exists = await check_flag_exists("ui_flag_for_o")
+    if flag_exists:
+        ui_flag_for_o = await sync_to_async(waffle.flag_is_active)(
+            request, "ui_flag_for_o"
+        )
+        user_flag_active = await sync_to_async(waffle.flag_is_active)(
+            request.user, "ui_flag_for_o"
+        )
+        if ui_flag_for_o or user_flag_active:
+            return await render_opinion_view(request, pk, "opinions")
+    return await view_opinion_old(request, pk, "str")
+
+
+async def view_opinion_pdf(
+    request: HttpRequest, pk: int, _: str
+) -> HttpResponse:
+    """View for displaying opinion case details."""
+    return await render_opinion_view(request, pk, "pdf")
+
+
+async def view_opinion_authorities(
+    request: HttpRequest, pk: int, _: str
+) -> HttpResponse:
+    """View for displaying opinion authorities."""
+    cluster: OpinionCluster = await aget_object_or_404(OpinionCluster, pk=pk)
+
+    authorities_context: AuthoritiesContext = AuthoritiesContext(
+        citation_record=cluster,
+        query_string=request.META["QUERY_STRING"],
+        total_authorities_count=await cluster.aauthority_count(),
+        view_all_url="view_authorities_url",
+        doc_type="opinion",
+    )
+    await authorities_context.post_init()
+
+    additional_context = {
+        "authorities_context": authorities_context,
+        "authorities_with_data": await cluster.aauthorities_with_data(),
+    }
+    ui_flag_for_o = await sync_to_async(waffle.flag_is_active)(
+        request, "ui_flag_for_o"
+    )
+    user_flag_active = await sync_to_async(waffle.flag_is_active)(
+        request.user, "ui_flag_for_o"
+    )
+
+    if ui_flag_for_o or user_flag_active:
+        return await render_opinion_view(
+            request, pk, "authorities", additional_context
+        )
+    else:
+        # Old page to load for people outside the flag
+        return await view_authorities(
+            request=request, pk=pk, slug="authorities"
+        )
```

Comment:
The [flag_is_active](https://waffle.readthedocs.io/en/stable/usage/views.html#usage-views) function takes two arguments, the `request`, and the name of a `flag`.

```suggestion
    ui_flag_for_o = await sync_to_async(waffle.flag_is_active)(
        request, "ui_flag_for_o"
    )
    if ui_flag_for_o:
        return await render_opinion_view(
            request, pk, "authorities", additional_context
        )
 
    # Old page to load for people outside the flag
    return await view_authorities(
        request=request, pk=pk, slug="authorities"
    )
```